src/database/schema/UserSchema.ts

```diff
@@ -178,6 +202,18 @@ class UserSchema implements BaseDbSchema {
     db.transaction(resolver);
   };
 
+  static async getSelfAnonUserInfo(
+    db: SQLiteDatabase,
+    selfAnonUserId: string,
+    channelId: string
+  ): Promise<UserSchema> {
+    const [results] = await db.executeSql(
+      `SELECT * FROM ${UserSchema.getTableName()} WHERE channel_id = ? AND user_id = ? LIMIT 1`,
+      [channelId, selfAnonUserId]
+    );
+    return results.rows.raw().map((dbObject) => this.fromDatabaseObject(dbObject))[0];
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new method `getSelfAnonUserInfo` is introduced to fetch anonymous user info. However, there's no error handling in case the database query fails or returns no results. It would be better to add some error handling here.
```diff
  static async getSelfAnonUserInfo(
    db: SQLiteDatabase,
    selfAnonUserId: string,
    channelId: string
  ): Promise<UserSchema> {
    try {
      const [results] = await db.executeSql(
        `SELECT * FROM ${UserSchema.getTableName()} WHERE channel_id = ? AND user_id = ? LIMIT 1`,
        [channelId, selfAnonUserId]
      );
+     if (results.rows.length === 0) {
+       throw new Error('No user found with provided ID and channel ID');
+     }
      return results.rows.raw().map((dbObject) => this.fromDatabaseObject(dbObject))[0];
+   } catch (error) {
+     console.error(error);
+     throw error;
+   }
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->